\chapter{Methods}\label{ch:methods}
With an understanding of the basics behind us, in this chapter we provide an overview of our contributions and proposed solution.

\section{Contributions}

\section{Experimental Designs} \label{experimental_design}
To assist in determining which conditions might lead to reductive evolution, we need to isolate individual variables and change just one thing at a time in order to see what effect, if any, it has on the final genome. To that end, we designed and conducted a series of experiments in which a ``wild type'' genome was allowed to evolve in differing conditions for 500,000 generations before analyzing the effects. To first create the wild type, a genome was generated randomly in Aevol which had at least one coding gene and which was allowed to evolve for 10 million generations in a non-varying environment. By the beginning of its use in our experiments, the wild type comprised 13,237 base pairs with 132 functional genes (i.e. genes which produce a gene product) and around 37\% non-coding DNA. 

In our experiments, we allowed the wild types to continue to evolve in the same environment in a total of 6 different conditions: with an increased/decreased selection strength, an increased/decreased mutation rate, and an increased/decreased population size.  For all experiments the environmental target function did not vary with respect to the time of the wild type generation nor with respect to any of the performed experiments. The three Gaussian functions are characterized by a height $h$, mean $m$, and width $w$:
\begin{equation*}
y(x) = h * e^{\left(\frac{-(x-m)^2}{2 * w^2} \right)}
\end{equation*}

An approximation of the target function used in our experiments, $f_E$, can be seen below in Figure~\ref{fig:target_function}. 

\begin{figure}[H]
	\centering
	\includegraphics[width=\linewidth]{environmental_target_function01}
	\caption[Experimental target function]{A visual approximation of the target function $f_E$ for our experiments.}
	\label{fig:target_function}
\end{figure}
 The parameters for the three Gaussians are given in the table below. 
\begin{table}[H]
	\centering
	\begin{tabular}{|c||c|c|c|}
		\cline{2-4}
		
		\multicolumn{1}{c||}{}& \textbf{h} & \textbf{m} & \textbf{w} \\
		\hline \hline
		Gaussian 1 & 1.2 & 0.52 & 0.12 \\
		\hline
		Gaussian 2 & -1.4 & 0.5 & 0.07 \\
		\hline
		Gaussian 3 & 0.3 & 0.8 & 0.03 \\
		\hline
	\end{tabular}
	\caption[Gaussian environmental parameters]{The Gaussian environmental target function parameters.}
	\label{table:gaussian_environment_params}
\end{table}
 
In addition to the six tested changed conditions, a control condition was performed in which the wild type was simply allowed to continue to evolve for the 500,000 generations with no change in any of the parameters from the generation of the wild type. To minimize bias, for each condition we performed five runs each (i.e. 5 rounds of mutation up, 5 rounds of mutation down, etc.), where each run had a differing random seed to control for the deterministic effects of the pseudorandom nature of Aevol's stochastic processes. This lead to a total of 35 experiments, all of which were carried out on a cluster from bwCloud\footnote{\url{https://www.bw-cloud.org/}}. 

The resulting data was processed using a combination of Python\footnote{\url{https://www.python.org/}}, Pandas\footnote{\url{https://pandas.pydata.org/}}, and Jupyter Notebooks\footnote{\url{https://jupyter.org/}}.  

\subsection{Inputs}
In Table~\ref{table:input_parameters}, our parameter values for all input parameters may be found. Please note that for $\mu$, this represents the mutation rates for point mutations, small insertions, and small deletions. The rearrangement rates were not changed under any condition and were always $1e-6$ for duplications, deletions, translocations, and inversions. 

\begin{table}[h]
	\centering
	\begin{tabular}{|c||c|c|c|}
		\hline
		 & \multicolumn{3}{c|}{\textbf{Parameter}} \\
		\cline{2-4}
		\textbf{Condition} &$\mu$ & $k$ & $N$ \\
		\hline
		control & $1.00E{-7}$ & $1000$ & $1024$ \\
		\hline
		mutation up & $4.00E{-7}$ & $1000$ & $1024$ \\
		\hline
		mutation down & $2.50E{-8}$ & $1000$ & $1024$ \\
		\hline
		selection up & $1.00E{-7}$ & $4000$ & $1024$ \\
		\hline
		selection down & $1.00E{-7}$ & $250$ & $1024$ \\
		\hline
		population up & $1.00E{-7}$ & $1000$ & $4096$ \\
		\hline
		population down & $1.00E{-7}$ & $1000$ & $256$ \\		
		\hline
	\end{tabular}
	\caption[Table of parameters]{Table of input parameters. $\mu$ is the mutation rate, $k$ is the selection strength, and $N$ is the population size.}
	\label{table:input_parameters}
\end{table}
As can be seen in Table~\ref{table:input_parameters}, only one parameter varied per condition in order to isolate potential influences on the outcome. A multiplier of $4$ was chosen for the differing conditions relative to the control condition (e.g. $N_\text{population up} = 4096 = 4*N_\text{control} = 4*1024$). This choice of a $4$x multiplier was chosen following the model of Carde et al.\cite{carde.2019}. In all conditions, the environment did not vary, and once the experiment was begun, the above parameters were held steady as well. 


\section{Evaluation Strategy}
In this section, we will examine the criteria we will be using to evaluate the results. The primary criteria will be examining the evolved genome's evolvability, robustness, and structure, with a statistical analysis of the changed conditions vs. the control condition. 

\subsection{Statistical Analysis of the Conditions}
We must first determine how to tell whether the results of the condition (e.g. mutation up, selection down, etc.) were significantly different from the control condition, statistically speaking. To do this, for some test variable (e.g. robustness, evolvability, etc.) we will first calculate the mean of all seeds for that condition and compare it to the mean across all seeds of the control condition. For the comparison,  we will use the ``Mann-Whitney U'' test. The Mann-Whitney U test is similar to the Wilcoxon signed-rank test, but it is used when the distribution of the two samples cannot be assumed to be normally distributed. The Mann-Whitney U test can also be used when the sample sizes are different, for example when the number of base pairs is different between two organisms. 

The purpose of the Mann-Whitney U test is to check whether two independent samples were selected from populations having the same distribution. Similar to other rank-sum tests, the test consists of the following steps:
\begin{enumerate}
	\item Assign numeric ranks to all observations;
	\item Add up the ranks for the observations from the first sample, giving us $R_1$
	\item The statistic $U_1$ for the first sample is given by:
	\begin{equation*}
	U_1 = R_1 - \frac{n_1*(n_1 + 1)}{2}
	\end{equation*}
	where $n_1$ is the sample size for the first sample and $R_1$ is the sum of the ranks from the first sample.
\end{enumerate}
The $U$ statistic for the second sample (i.e. $U_2$) is computed analogously. We then use $U = U_1 + U_2$ which, for sample sizes greater than 20, is approximately normally distributed. Then the standard score is given by:
\begin{equation*}
z = \frac{U - m_U}{\sigma_U}
\end{equation*}
where $m_U$ and $\sigma_U$ are the mean and standard deviation of $U$.
\begin{equation*}
m_U = \frac{n_1*n_2}{2} \text{and } \sigma_U = \sqrt{\frac{n_1*n_2\left(n_1 + n_2 +1\right)}{12}}
\end{equation*} We used the Python library function \texttt{scipy.stats.mannwhitneyu} to calculate this statistic, which also calculates the p-value. If the p-value is below $0.05$, we may reject the null hypothesis $H_o$ that the two samples (i.e. the control and the condition) are from the same distribution.

\subsection{Fitness}
Fitness in Aevol is closely tied in to the ``metabolic error''. This error, $g$, is calculated as the gap between the environmental function $f_E$ and the phenotype of the organism, $f_P$. Once $g$ is determined as described above in Section~\ref{subsec:aevol_selection}, it may be used to calculate the actual fitness of the organism according to the equation:
\begin{equation*}
\text{fitness} = exp(-k*g)
\end{equation*} 
where k is the \textit{selection coefficient} variable set by the user in a parameter file. Aevol provides fitness statistics both for the fittest individual and for the population at large.
\subsection{Robustness}
Aevol calculates statistics for both mutational robustness as well as antirobustness. Robustness in Aevol is calculated similarly to evolvability: a \texttt{lineage} file for an individual is fed to the post-treatment \texttt{misc\_ancestor\_robustness}, which generates a large number (specifiable by the user) of offspring, whose fitness is then measured. The percentage of these offspring which are neutral (i.e. whose phenotype is not affected by the mutations) is the robustness, and the percentage of positive offspring determines the antirobustness.

In our experiments, at the end of the run of 500,000 generations we generated a lineage file for the best individual at generation 500,000, i.e. the individual whose metabolic error was smallest. This lineage file is then fed in to the post-treatment \texttt{aevol\_misc\_ancestor\_robustness} and robustness statistics are generated for each generation in the lineage file. The statistics produced by this post-treatment are summarized in Table~\ref{table:robustness} below.
\begin{table}[H]
	\centering
	\begin{tabular}{||c||}
		\hline
		\textbf{Statistic} \\
		\hline \hline
		Fraction of positive offspring \\
		\hline
		Fraction of neutral offspring (aka reproductive robustness) \\
		\hline
		Fraction of neutral mutants (aka mutational robustness) \\
		\hline
		Fraction of negative offspring \\
		\hline
		Cumulative delta-gaps of positive offspring \\
		\hline
		Cumulative delta-gaps of negative offspring \\
		\hline
		Delta-gap for the best offspring \\
		\hline
		Delta-gap for the worst offspring \\
		\hline
		Cumulative delta-fitness of positive offspring \\
		\hline
		Cumulative delta-fitness of negative offspring \\
		\hline
		Delta-fitness for the best offspring \\
		\hline
		Delta-fitness for the worst offspring \\
		\hline
		
	\end{tabular}
	\caption[Aevol robustness statistics]{Table of robustness statistics calculated by Aevol for the best individual with the provided lineage.}
	\label{table:robustness}
\end{table}
We may then compare these statistics for both the control and specific condition we wish to compare (e.g. mutation up). Because this data is somewhat noisy (owing to the fact that the fitness may change rapidly) we will use box and whisker plots to show the overall spread rather than plotting the robustness generation by generation. 
\subsubsection{Evolvability}
In Aevol, evolvability is calculated by generating a large set of offspring for a specific individual (one whose lineage was generated using the post-treatment \texttt{aevol\_misc\_lineage}) at regular periods along their lineage and then determining the number of ``positive offspring''. Positive offspring are defined as those whose fitness is greater than their parent's. The evolvability of an individual is then the sum total of all improvement of all of the beneficial offspring, i.e.:

\begin{equation*}
\text{evolvability} = \frac{|\text{positive offspring of }i|}{|\text{total offspring of }i|}*\sum \Delta^{\text{positive offspring}}_{fitness}
\end{equation*}  where $\Delta^{\text{positive offspring}}_{fitness}$ is the cumulative sum of the fitness increase for the positive offspring. Thus, evolvability in Aevol accounts for both the likelihood of a positive mutation and the average improvement provided by said mutation. Practically speaking, to find an organisms evolvability in Aevol one must give the post-treatment \texttt{misc\_ancestor\_robustness} a lineage file and then multiply the fraction of the number of positive offspring (column 2) by the cumulative total of the fitness gap $g$ of the positive offspring (column 10).

%\subsubsection{Time to Coalescence}

 
\subsection{Structure}\label{methods:structure}
Another strength of Aevol is its ability to analyze changes in the structure of DNA and RNA. As with fitness, Aevol produces statistics about individuals and the population at large for many aspects of genome structure, including: the number of coding vs. non-coding bases (i.e. they respectively do or do not code for at least one protein), the average size of coding and non-coding DNAs, the number of genes, the number of ``essential'' base pairs (i.e. those that are part of a functional coding sequence), etc. Tables~\ref{table:aevol_stats_genes_and_bp} and Table~\ref{table:aevol_stats_fitness_and_mutation} summarize the different statistics and where they are to be found Aevol's output files. 

\begin{table}[H]
	\centering
	\begin{tabular}{ |c|c| }
		\hline
		\multicolumn{2}{|c|}{\textbf{Stat File}} \\
		\hline
		\texttt{stat\_genes\_}$\langle$\texttt{best/glob}$\rangle$ & \texttt{stat\_bp\_}$\langle$\texttt{best/glob}$\rangle$ \\
		\hline \hline
		number of coding RNAs & number of bp not in any CDS \\
		\hline
		number of non-coding RNAs & number of bp not included in any functional CDS \\
		\hline
		average size of coding RNAs & number of bp not included in any non-functional CDS \\
		\hline
		average size of non-coding RNAs & number of bp not included in any RNA \\
		\hline
		number of functional genes & number of bp not included in any coding RNA \\
		\hline
		number of non-functional genes & number of bp not included in any non-coding RNA \\
		\hline
		average size of functional genes & number of non-essential bp \\
		\hline
		average size of non-functional genes & number of non-essential bp including non-functional genes \\
		\hline
	\end{tabular}
	\caption[Aevol's stats: genes and base pairs]{Statistics found in \texttt{stat\_genes} and \texttt{stat\_bp} output files from Aevol. $\langle$best/glob$\rangle$ indicates that statistics are available for both the best individual and the average across the whole population.}
	\label{table:aevol_stats_genes_and_bp}
\end{table}
``Essential'' base pairs are those whose mutation would change the phenotype of the organism.
\begin{table}[H]
	\centering
	\begin{tabular}{ |c|c| }
		\hline
		\multicolumn{2}{|c|}{\textbf{Stat File}} \\
		\hline
		\textbf{stat\_fitness\_$\langle$best/glob$\rangle$} &
		\textbf{stat\_mutation\_$\langle$best/glob$\rangle$} \\
		\hline \hline
		population size & number of local mutations \\
		fitness & number of chromosomic rearrangements \\
		genome size & number of switches \\
		metabolic error & number of indels \\
		parent's metabolic error & number of duplications \\
		metabolic fitness & number of deletions \\
		secretion error & number of translocations \\
		parent's secretion error & number of inversions \\
		secretion fitness & \\ 
		amount of compound present in grid-cell & \\
		\hline
	\end{tabular}	
	\caption[Aevol's stats: fitness and mutation]{Statistics found in \texttt{stat\_fitness} and \texttt{stat\_mutation} files from Aevol. $\langle$best/glob$\rangle$ indicates that statistics are available for both the best individual and the average across the whole population.}
	\label{table:aevol_stats_fitness_and_mutation}
\end{table}

\section{Expected Results}\label{sec:expected_results}
Given the state of the literature and other experiments, as partially described in Section~\ref{related_work}, the table below summarizes our expected results. 

\begin{table}[H]
	\centering
	\begin{tabular}{|c||c|c|c|c|c|c|}
		\hline
		\multicolumn{7}{|c|}{{\Large \textbf{Experiment Predictions}}} \\
		\hline \hline
		\multirow{2}{*}{\textbf{Result}} & \multicolumn{6}{c|}{\textbf{Condition}} \\
		\cline{2-7}
		 & {\Large$\mu_+$} & {\Large$\mu_-$} & {\Large$k_+$} & {\Large$k_-$} & {\Large$N_+$} & {\Large$N_-$} \\
		\hline 
		Genome Size & $-^{\cite{bradwell2013correlation, marais2008mutation}}$ & $\text{+}^{\cite{bradwell2013correlation, drake1991constant}}$ & $+^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2014}}$ & $+^{\cite{Batut.2014}}$ \\
		\hline
		Fitness & $+^{\cite{bataillon2000estimation, vahdati2017effect}}$ & $+^{\cite{vahdati2017effect}}$ & $+^{\cite{Batut.2014}}$ & $-^{\cite{Batut.2014}}$ & $+^{\cite{cutter2019primer, vahdati2017effect}} $ & $-^{\cite{cutter2019primer, vahdati2017effect}} $\\
		\hline
		Amount of non-coding DNA & $-^{\cite{Knibbe2007}}$ & $+^{\cite{Knibbe2007}}$ & $+^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2013}}$ & $+^{\cite{Batut.2013}}$ \\
		\hline
		Number of genes & $-^{\cite{Knibbe2007}}$ & $+^{\cite{Knibbe2007}}$ & $-^{\cite{Knibbe2007}}$ & $+^{\cite{Knibbe2007}}$ & $-^{\cite{Batut.2014}}$ & $+^{\cite{Batut.2014}}$ \\
		\hline
		Average size of genes & $-^{\cite{Liard.2018}}$ & $+^{\cite{Liard.2018}}$ & $-^{\cite{Batut.2013}}$ & $+^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2014}}$ & $+^{\cite{Batut.2014}}$ \\
		\hline
		Robustness & $-^{\cite{Knibbe2007}}$ & $+^{\cite{Knibbe2007}}$ & $-^{\cite{Batut.2013}}$ & $+^{\cite{Batut.2013}}$ & $-^{\cite{elena2007effects}}$ & $+^{\cite{elena2007effects}}$ \\
		\hline
		Evolvability & $+^{\cite{Knibbe2007}}$ & $-^{\cite{Knibbe2007}}$ &  $+^{\cite{Batut.2013}}$ & $-^{\cite{Batut.2013}}$ & $-^{\cite{wein2019effect}}$ & $+^{\cite{wein2019effect}}$ \\
		\hline		
	\end{tabular}
	\caption[Experiment expectations]{Our predictions for the experiments. $\mu$ is the mutation rate, ~$k$ is the selection rate, and $N$ is the population size. $\mu_+$ indicates an increased mutation rate, $\mu_-$ a decreased rate, etc. A $+$ in the main grid space indicates we expect an increase (over the control condition) for that condition, and a $-$ indicates an expected decrease for that condition.}
	\label{table:experiment_predictions}
\end{table}


