\pagenumbering{arabic}%DO NOT REMOVE THIS
\chapter{Introduction}\label{ch:intro}

One way of speaking about evolution is with regard to the change in frequencies of alleles (variants of genes) in a population over time. In the field of population genetics, the \textit{Hardy-Weinberg principle} states that genetic variation (diversity of gene frequencies) in a population will remain constant from generation to generation in the absence of disturbances. There are five ``Hardy-Weinberg assumptions'' which must hold for this to be true: no mutation, random mating, no gene flow (i.e. the transfer of genetic material from one population to another), infinite population sizes, and no selection~\cite{cutter2019primer}. If any of these assumptions does not hold, then this may change the frequencies of alleles in a population, which allows evolution to occur. The mechanisms for evolution are, then, the opposite of these assumptions, namely: mutation, non-random mating, gene flow, finite populations, and natural selection. Pulling at any of these levers has an effect on the genetic variation of a population and thus impacts their genome.  

Reductive evolution is the process of the average genome size of a species shrinking over time, with respect to both the number of base pairs and genes. For example, some species of bacteria have experienced reductive evolution over the course of millions of years, and this reduction in their genome has lead to a loss of genes, certain regulatory abilities, and more. One such bacterial species is the marine cyanobacteria \textit{Prochlorococcus}, some of whose strains have experienced a reduction of nearly 40\% of their base pairs when compared to larger strains of their closest living relative, \textit{Synechococcus}. Even among strains of Prochlorococcus, the difference between some of the larger and smaller strains is upwards of 38\% \cite{Batut.2014}. Despite being extensively studied, the underlying mechanisms and full impact of reductive evolution are not fully understood and are an area of intense current research. Several competing explanations exist for these mechanisms, from the influence of population size, genetic drift (defined below), or selection \cite{Batut.2014}.

Although it would provide more conclusive evidence, performing \textit{in vivo} experiments with living organisms is often impractical because of the difficulty or impossibility of reproducing natural environmental conditions in a laboratory. Such experiments are often too costly in terms of both time and resources. As an alternative, \textit{in silico} experimental evolution is one option that can be used to study the conditions under which an organism's genome may become reduced. In this method, organisms and their evolution over thousands or millions of generations are simulated in software. In this manner, one can control and evaluate every aspect of their evolution over time and a full record of their lineage may be maintained and studied, allowing one to go back and closely examine every step of the evolutionary period for a greater understanding of the factors that lead to specific effects on the genome. The in silico tool \textit{Aevol} is one such platform which realistically models bacterial genomes and evolution, allowing one to draw conclusions about their real-world counterparts. In the following thesis, the results of experiments in artificial evolution are presented which aim to identify and evaluate several factors which potentially lead to changes in genome structure and a reduced genome in simulated bacteria using the Aevol platform. 

\section*{Problem Statement} \label{problem_statement}
Among the difficulties of studying reductive evolution with in vivo evolutionary experiments, one of the most difficult obstacles to overcome is the lack of a full ancestral record. This lack of a full phylogeny can make it difficult or impossible to tell exactly when and how a specific event occurred, or a trait evolved or was lost, as illustrated in Figure \ref{fig:phylogeny03} below. 
\begin{figure}[h]
\includegraphics[scale=0.75]{phylogeny03}
\centering
\caption[Unknown phylogeny]{An illustration of unknown phylogeny. Since the phylogenetic information under the shaded box is typically not known, the point of divergence (red circle) can't be determined.}
\label{fig:phylogeny03}
\end{figure}
In this example, two related organisms A and B are compared with an attempt to determine when and how a specific trait was gained or lost by one of the organisms. This may be useful, for example, when attempting to estimate the relative importance (due to conservation over many generations) of some trait. Without the phylogenetic information (under the shaded box) it may not be possible to identify the point in their evolutionary history at which the two organisms diverged, making time estimates difficult or impossible.

Another major downside to in vivo evolutionary experiments is that they are slow. For example, the well-known E. coli Long-Term Evolution Experiment (LTEE) by Profesor Lenski at Michigan State University has been ongoing since February of 1988 and only passed generation 50,000 in 2010, 22 years later\footnote{\url{http://myxo.css.msu.edu/ecoli/celebrate50K.html}}. 

As an alternative to in vivo experiments, in silico evolutionary experiments are well-suited to the task of studying reductive evolution. Generations of organisms may be evolved within a very short time period, and a full "fossil record" of each lineage may be kept on disk for further analysis. 

The in silico tool Aevol has a realistic artificial chemistry model which was developed specifically to study genome structure. It contains tools to analyze the robustness, fitness, and evolvability of digital organisms over time.  In this thesis, Aevol is used to perform experiments in silico evolution in order to determine the effects of several different conditions on the genome of a simulated ``wild type'' bacteria.

\subsection{Genetic Variability}
%TODO How genome size evolves, the models (Lynch, Moran, etc.) - FLESH THIS OUT!
Of fundamental importance to this thesis is the question of how genome size evolves. Several models have been proposed over the years, and two which have proven to be particularly useful are the Moran and Lynch models. These two models relate to the question of genetic variability, i.e. the genetic differences  

Michael Lynch, the famous geneticist currently at Arizona State University has a model for determining genetic variabiltiy which is still commonly in use now and is known as the Lynch model. In this model, 
%TODO Complexity of evolution mechanisms

%TODO Hypothesis for reductive genome evolution
%TODO Comparative genomics
%TODO What is interesting about reductive evolution?
%TODO Why is aevol not realistic?
%TODO Are there any other in silico experimental evolution tools?
\section*{Report outline}
%TODO Probably take this out
This chapter serves as the introduction to the thesis and the research problem being faced. In Chapter~\ref{ch:background}, some necessary background information is provided on reductive evolution, in silico evolution in general, and Aevol in particular. Chapter~\ref{ch:methods} describes the experimental setup. Chapter~\ref{ch:results_discussion}
provides the results and analysis of the experiments of this thesis, and Chapter~\ref{ch:conclusion} presents the conclusions drawn from this work. 


